File: searches/linear_search.js
/**
 * Linear Search implementation in JavaScript
 * @param {Array} arr - Array to search
 * @param {number} target - Value to search for
 * @returns {number} - Index of target or -1
 */
function linearSearch(arr, target) {
    for (let i = 0; i < arr.length; i++) {
        if (arr[i] === target) {
            return i;
        }
    }
    return -1;
}

// Example usage
const arr = [10, 50, 30, 70, 80, 20];
const target = 30;
const result = linearSearch(arr, target);

console.log(`Element found at index: ${result}`);

File: searches/linear_search.go
package main

import "fmt"

// LinearSearch returns the index of target in arr, or -1 if not found
func LinearSearch(arr []int, target int) int {
	for i, v := range arr {
		if v == target {
			return i
		}
	}
	return -1
}

func main() {
	arr := []int{10, 50, 30, 70, 80, 20}
	target := 30
	result := LinearSearch(arr, target)
	
	if result != -1 {
		fmt.Printf("Element found at index %d\n", result)
	} else {
		fmt.Println("Element not found")
	}
}
